content/getting-started/configuring-your-local-environment/downloading-and-installing-node-js-and-npm.mdx

```diff
@@ -30,6 +31,10 @@ node -v
 npm -v
 ```
 
+## Using Npm to install Node locally
+
+You can `npm i node@version` to install a specific version of node. Npm will use the installed version when running any scripts defined in `package.json#scripts`. This is recommended because it lets you lock a particular project to a specific version of Node without affecting the rest of your system
```

Comment:
This is not a way to install the Node.js that I would like us advocated as npm.

Reply:
I guess I don't understand how this way is at all worse than the suggested `nvm` way. It's easier, faster, more reliable, and reproducible. `nvm` doesn't run on Windows. What does `nvm` have that `npm i --save node` doesn't?

Or, at least, why is "nvm" the "strongly recommended" way? I see the explanation about npm permissions but that's only a problem if you're installing global packages, which isn't the default and requires elevated privileges anyway. If it's not recommended, why is this the default Node.js has stuck with? I missing something?

Reply:
> I guess I don't understand how this way is at all worse than the suggested `nvm` way. It's easier, faster, more reliable, and reproducible. `nvm` doesn't run on ~~Windows~~ non-POSIX. What does `nvm` have that `npm i --save node` doesn't?
> 
> Or, at least, why is `nvm` the "strongly recommended" way? I see the explanation about Npm permissions but, as far as I understand, that's only a problem if you're installing global packages, which isn't the default and requires elevated privileges anyway. If it's not recommended, why is this the default Node.js has stuck with? I missing something?

I worked it out I think

Reply:
installing Node.js has some shortcomings such as "npm install" and initial npm scripts running with a different version of npm / node.js then subsequent scripts without making changes to your system path.

If you try to run `node` or `npm` your system will first attempt to use the version in your path. It is only npm scripts that inject `node_modules/.bin` into your path. So if you were to do something like run `npm test` that will initially use your system version of node.js / npm.

These kinds of inconsistencies open up for a large number of possible edge cases and inconsistencies

Reply:
@MylesBorins That is a good point, but is a known issue. This only affects the *first* compilation on every system which is rather manageable, imho. It also only affects repos that "opt-in" to using the `node` package on Npm. Usually we don't need to lock `node` to a particular version, or at least `package.json#engines` is usually adequate in my experience. Note, once installed, `npm test` **will** use the correct version of `node` for actually **running** the tests. The top level node that is running npm may be different, but that is *desired*. What if I want to use modern versions of Npm that only work on recent versions of Node but run a project with an older version of Node?

This also ignores the issues with `nvm`

- It's a 3rd party package (yes, the `node` package is as well, but not everyone needs it)
- It doesn't work on Windows directly (without WSL or msys or needs `nvm-windows`...)
- It affects the other projects (I can't easily work on two projects with different node versions at the same time and messing with PATH is something I think should be avoided, especially on Windows)
- Needs extra configuration (`.nvmrc`) to ensure correct version is easily installed

Reply:
> that is a good point, but is a known issue. This only affects the first compilation on every system which is rather manageable, imho

This is where we disagree. This is not an acceptable trade off in my opinion and I don't think we should be suggesting this solution in the npm documentation.